### PR TITLE
Fix parsing of unclean map values in Config.

### PR DESCRIPTION
**Pull request by anuraaga**
We currently reject map values where elements within them are e.g., empty, but we should ignore those (it's easy to have empty values, trailing commas, etc when creating a variable in a shell script).

I just copied the implementation as is from SDK.

Also added the ConfigProperties "TCK" from the SDK :)

### CODE REVIEW

File: javaagent-tooling/src/test/java/io/opentelemetry/javaagent/tooling/config/ConfigPropertiesAdapterTest.java [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    assertThat(config.getInt("int")).isNull();
+    assertThat(config.getLong("long")).isNull();
+    assertThat(config.getDouble("double")).isNull();
+    assertThat(config.getCommaSeparatedValues("list")).isEmpty();
+    assertThat(config.getCommaSeparatedMap("map")).isEmpty();
+    assertThat(config.getDuration("duration")).isNull();
+  }
+
+  @Test
+  @Disabled("Currently invalid values are not handled the same as the SDK")
```
Comment (anuraaga) [inline javaagent-tooling/src/test/java/io/opentelemetry/javaagent/tooling/config/ConfigPropertiesAdapterTest.java:81]: @mateuszrzeszutek I think we should make sure the behavior is the same as the SDK for these too

File: instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/config/ConfigValueParsers.java [hunk 2]
```diff
@@ -6,36 +6,45 @@
...
-    }
-    return Collections.unmodifiableMap(result);
+    return parseList(value).stream()
+        .map(keyValuePair -> filterBlanksAndNulls(keyValuePair.split("=", 2)))
+        .map(
+            splitKeyValuePairs -> {
+              if (splitKeyValuePairs.size() != 2) {
+                throw new IllegalArgumentException(
+                    "Invalid map property, should be formatted key1=value1,key2=value2: " + value);
+              }
```
Comment (trask) [inline instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/config/ConfigValueParsers.java:32]: should we allow empty values? e.g. `key1=val1,key2=,key3=val3`

**Review by trask — APPROVED** [review]

File: instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/config/ConfigValueParsers.java (hunk 2 above)
Comment (anuraaga) [reply instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/config/ConfigValueParsers.java:32]: We do for `getString` (at least in SDK) though seem to be missing tests for it, and the map with empty values. But I think returning empty when it's empty is reasonable behavior, it's similar to how query params work

**Review by mateuszrzeszutek — APPROVED** [review]

File: javaagent-tooling/src/test/java/io/opentelemetry/javaagent/tooling/config/ConfigPropertiesAdapterTest.java (hunk 1 above)
Comment (mateuszrzeszutek) [reply javaagent-tooling/src/test/java/io/opentelemetry/javaagent/tooling/config/ConfigPropertiesAdapterTest.java:82]: Yeah, that'd certainly make sense to do so. On the other hand, in the agent code I think we'd probably prefer to fall back on the default value instead of throwing an exception in the middle of an advice class. So, how about the following:
* in the `@Nullable X getX(String name)` methods (these are used only in the SDK bridge, I think) we'll throw an exception if the value is invalid;
* but in the `X getX(String name, X default)` (which are used pretty much everywhere in the agent) we'll return the default value in case of any errors

Comment (anuraaga) [reply javaagent-tooling/src/test/java/io/opentelemetry/javaagent/tooling/config/ConfigPropertiesAdapterTest.java:82]: Yeah I think that's fine
